content/projects/data-science-specific/data-wrangling/_index.md

```diff
@@ -79,25 +80,24 @@ df.apply(get_percentage, axis = 1) #axis=1 applies the function to all columns
 ├──requirements.txt
 └──.gitignore 
 ```
-2. Import the dataset [personality_scores.csv](personality_scores.csv). This data should be named `personality_df` Examine the DataFrame for duplicates (based on ID), and drop any duplicates that exist. Use an assert statement to check that the DataFrame is the length of the unique entries of the original data frame. This new DataFrame should be named `personality_score_df`. If there are empty columns, please ensure you remove them.
+2. Import the dataset [personality_scores.csv](personality_scores.csv) and load it into a DataFrame. Examine the DataFrame for duplicates (based on ID), and drop any duplicates that exist. Let's imagine that the data is ordered chronologically, and we'd like to keep the data related to the most recent time an applicant did the test. From your recent project on [assertive programing tricks](projects/data-science-specific/assertive-programming-tricks-for-pandas/_index.md) use the methods you practiced to check that the updated DataFrame has the dimensions you expect it to have. Perform some final cleaning steps by removing empty columns and cleaning up column names. Store this updated version of the DataFrame in a variable called: `personality_score_df`.
 
-Tip: An example assert statement is `assert 2*20=40` and it's a great way to check that your modification of the data was successful.
 
-2. Create new columns containing the total score of each of the personality test subscales. The new columns should be named - `Conscientiousness`, `Emotional_Stability`, `Openness`, `Agreeable`, `Extraversion`. To do this, write a function (or functions) that will calculate the total score for each of the subscales (conscientiousness, emotional stability, openness to new experience, agreeableness, extraversion), as set out in [scoring](scoring.txt). In other words, for the conscientiousness total score, all items marked as belonging to 'conscientiousness' should be summed. You should also clean up the column names like this `Section 5 of 6 [I am always prepared.]` to something like this `I am always prepared`.
+3. Creat a function (or functions) to support creating new columns containing the total score of each of the personality test subscales. The new columns should be named as per the **big five personality traits** laid out in [scoring](scoring.txt). In other words, for the **conscientiousness** total score, all items marked as belonging to 'conscientiousness' should be summed.
```

Comment:
If we want to automark things according to the column names then we should tell learners a bit more. Like, at least tell them the naming convention we will be following. I dont think they are likely to guess right. We should never set the learners up so that they need to make guesses in order to pass the automarker's tests.

And then we should be precise about the names we use is we use them. I'd recommend using pep8 style naming. Also it would be good to be consistent with the words used, eg "openness" and "agreeable" seems like a bit of a funny choice.

Reply:
Ja as I mentioned in chat, I was trying to get learners to have to go find the new column names in the data provided, but I'll change it back. 
This is the line we tread in trying to not give them everything on a platter but wanting them setup for success with the automarker

- the word choice was taken from the data provided for what the personality traits are, so I am confused by why those are funny choices?

Reply:
The naming style is also an interesting consideration since we already have a bunch of columns named, for example like: `I am always prepared`. So do we insist they convert all those to pep8 style too? Else if we then insist on pep8 for new columns the naming across the df will be inconsistent - that is why I didn't specify that for the new columns